src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp

```diff
@@ -82,20 +85,68 @@ getValue(T_Type& value)
 
 }
 
+/** set a value to all elements of a box
+ *
+ * @tparam T_numWorkers number of workers
+ * @tparam T_xChunkSize number of elements in x direction to prepare with one cuda block
+ */
+template<
+    uint32_t T_numWorkers,
+    uint32_t T_xChunkSize
+>
 struct KernelSetValue
 {
-    template <class DataBox, typename T_ValueType, typename Space>
-    DINLINE void operator()(DataBox data, const T_ValueType value, const Space size) const
+    /** set value to all elements
+     *
+     * @tparam T_DataBox PMacc::DataBox, type of the memory box
+     * @tparam T_ValueType type of the value
+     * @tparam T_SizeVecType PMacc::math::Vector, index type
```

Comment:
This is defined for the parameter `size`, this is only the type with no domain relation